src/editors/data/redux/thunkActions/video.js

```diff
@@ -1,13 +1,24 @@
 import { actions, selectors } from '..';
-import { removeItemOnce } from '../../../utils';
+import { formatDuration, removeItemOnce } from '../../../utils';
 import * as requests from './requests';
 import * as module from './video';
 import { valueFromDuration } from '../../../containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks';
 import { parseYoutubeId } from '../../services/cms/api';
 
-export const loadVideoData = () => (dispatch, getState) => {
+export const loadVideoData = (selectedVideoId) => (dispatch, getState) => {
   const state = getState();
-  const rawVideoData = state.app.blockValue.data.metadata ? state.app.blockValue.data.metadata : {};
+  let rawVideoData = state.app.blockValue.data.metadata ? state.app.blockValue.data.metadata : {};
+  if (Object.keys(rawVideoData).length === 0 && selectedVideoId !== null) {
+    const rawVideos = Object.values(selectors.app.videos(state));
+    const selectedVideo = rawVideos.find(video => video.edx_video_id === selectedVideoId);
+    // TODO it's missing laod the transcripts
+    rawVideoData = {
+      edx_video_id: selectedVideo.edx_video_id,
+      thumbnail: selectedVideo.course_video_image_url,
+      end_time: formatDuration(selectedVideo.duration),
```

Comment:
Don't set the end time here - just leave it empty on the editor page. Otherwise the UI will display "Custom Duration" instead of "Full video"

Reply:
Done